lib/shoryuken/manager.rb

```diff
@@ -58,24 +54,28 @@ def stop(options = {})
     def processor_done(queue)
       logger.debug { "Process done for '#{queue}'" }
 
-      dispatch
+      dispatch_async
     end
 
     private
 
-    def dispatch
-      return if @done.true?
-
-      @dispatcher_pool.post(&method(:dispatch_now))
+    def dispatch_async
+      @dispatcher_executor.post(&method(:dispatch_now))
     end
 
     def dispatch_now
-      return if ready.zero?
-      return unless (queue = @polling_strategy.next_queue)
+      return if @done.true?
+
+      begin
+        return if ready.zero?
+        return unless (queue = @polling_strategy.next_queue)
 
-      logger.debug { "Ready: #{ready}, Busy: #{busy}, Active Queues: #{@polling_strategy.active_queues}" }
+        logger.debug { "Ready: #{ready}, Busy: #{busy}, Active Queues: #{@polling_strategy.active_queues}" }
 
-      batched_queue?(queue) ? dispatch_batch(queue) : dispatch_single_messages(queue)
+        batched_queue?(queue) ? dispatch_batch(queue) : dispatch_single_messages(queue)
+      ensure
+        dispatch_async
```

Comment:
> sleep MINIMUM_WAIT # 0.05 or something to prevent CPU pegging

@waynerobinson I need to keep it running longer, but it seems to be working without a `sleep`. As the `@dispatcher_executor` discards `post` when there's a thread running, the request for fetching is kind of doing `sleep`, it should only be like without any sleep if all processors are busy or non queue is available as you pointed out. I will keep monitoring the CPU, if that does not work, we can def add `sleep 0.05`.